boot/kernel_os_test.go

```diff
@@ -471,54 +471,113 @@ func (s *ubootSuite) forceUbootBootloader(c *C) {
 	c.Assert(osutil.FileExists(fn), Equals, true)
 }
 
-func (s *ubootSuite) TestExtractKernelAssetsAndRemoveOnUboot(c *C) {
-	files := [][]string{
-		{"kernel.img", "I'm a kernel"},
-		{"initrd.img", "...and I'm an initrd"},
-		{"dtbs/foo.dtb", "g'day, I'm foo.dtb"},
-		{"dtbs/bar.dtb", "hello, I'm bar.dtb"},
-		// must be last
-		{"meta/kernel.yaml", "version: 4.2"},
-	}
+// forceUbootBootloader sets up a uboot bootloader, in the uc20 style where we
+// have a separate boot.sel file for snapd specific bootloader env
+func (s *ubootSuite) forceUC20UbootBootloader(c *C) {
+	bootloader.Force(nil)
 
-	si := &snap.SideInfo{
-		RealName: "ubuntu-kernel",
-		Revision: snap.R(42),
+	// to find the uboot bootloader we need to pass in NoSlashBoot because
+	// that's where the gadget assets get installed to
+	installOpts := &bootloader.Options{
+		NoSlashBoot: true,
 	}
-	fn := snaptest.MakeTestSnapWithFiles(c, packageKernel, files)
-	snapf, err := snap.Open(fn)
+
+	mockGadgetDir := c.MkDir()
+	// this must be empty for uc20 behavior
+	// TODO:UC20: update this test for the new behavior when that is implemented
+	err := ioutil.WriteFile(filepath.Join(mockGadgetDir, "uboot.conf"), nil, 0644)
+	c.Assert(err, IsNil)
+	err = bootloader.InstallBootConfig(mockGadgetDir, dirs.GlobalRootDir, installOpts)
 	c.Assert(err, IsNil)
 
-	info, err := snap.ReadInfoFromSnapFile(snapf, si)
+	// in reality for uc20, we will bind mount <ubuntu-boot>/uboot/ubuntu/ onto
+	// /boot/uboot, so to emulate this at runtime for the tests, just set
```

Comment:
```suggestion
	// /boot/uboot, so to emulate this at runtime for the tests, just
```

Reply:
mmm can't apply the suggestion, but committing it locally, thanks for the suggestion